overlay/default.nix

```diff
@@ -12,6 +12,48 @@ in
   # Uses sources.mix-to-nix and sources.gitignore
   nixUnstable = inputs.nix.defaultPackage.${final.system};
 
+  benchwrapper = prev.writeShellScriptBin "benchwrapper" ''
+    set -euo pipefail
+
+    # number of cores to run the wrapped program on
+    test_cpus=2
+
+    sctl=${prev.systemd}/bin/systemctl
+
+    #############
+
+    ncpus=$(( $(${prev.util-linux}/bin/lscpu -e | wc -l) - 1 ))
+
+    fullrange="0-$(( $ncpus - 1 ))"
+    sysrange="0-$(( $ncpus - $test_cpus - 1 ))"
+    shieldrange="$(( $ncpus - $test_cpus ))-$(( $ncpus - 1 ))"
+
+    cleanup() {
+      echo "called cleanup"
+      # restore full core utilization for all services
+      $sctl set-property --runtime system.slice AllowedCPUs="$fullrange"
+      $sctl set-property --runtime user.slice AllowedCPUs="$fullrange"
+    }
+
+    # make sure to allow scheduling on all cores again in case of error
+    trap cleanup EXIT
+
+    # limit user and system services to certain cores
+    $sctl set-property --runtime system.slice AllowedCPUs="$sysrange"
+    $sctl set-property --runtime user.slice AllowedCPUs="$sysrange"
+
+    # run this process and all children of it in the shield slice on the cores
+    # note: requires superuser rights to write here
+    mkdir -p /sys/fs/cgroup/shield
+    echo "$shieldrange" > /sys/fs/cgroup/shield/cpuset.cpus
+    echo ''$$ > /sys/fs/cgroup/shield/cgroup.procs
```

Comment:
Will it also work for all child processes invoked by `$$`? :thinking:

Reply:
according to the systemd manual all child processes inherit the parent's slice and would thereby run in the wrapper
I was looking into `systemd-run` to better handle this whole block